### PR TITLE
selinux: Fix error reporting

### PR DESCRIPTION
**Pull request by martinpitt**
Exception objects are not directly printable, so trying to render it in    
an Alert or with `console.warn()` just results in an empty string. That    
makes debugging really hard, as there is no corresponding entry in the    
journal either.    
    
Use `.toString()` on the exception to render them properly. Also show    
the actual exception text in the Alert in the UI. Still keep the    
console.warn()s, so that we can do naughty matching on them.    
    
While at it, eliminate the unnecessary Promise wrappers, and use chaining.    

----

I'm trying to get to the bottom of [this common flake](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18661-20230418-173000-028eaf96-centos-8-stream-pybridge/log.html#333). ([another instance](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18647-20230417-063552-d3f14f3e-rhel-8-9/log.html#139))

The [screenshot](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18661-20230418-173000-028eaf96-centos-8-stream-pybridge/TestSelinux-testTroubleshootAlerts-centos-8-stream-127.0.0.2-2801-FAIL.png) just says "Unable run fix:". I didn't yet reproduce this locally, but when I run the test in a loop, it sometimes fails to delete the alert. The UI alert just says that (without a reason), and the console says:
```
warning: Unable to delete alert with id 3030af65-8f20-4e46-8d88-2fd542cfae6e
warning:
```
So as the first step, let the page tell us what's actually wrong!

### CODE REVIEW

**Comment by martinpitt** [comment]
I locally got another "failed to delete". It says "message recipient disconnected from message bus without replying", in other words, setroubleshootd crashed. The journal just says:

```
Apr 19 08:28:48 centos-8-stream-127-0-0-2-2201 setroubleshoot[14254]: AnalyzeThread.run(): Set alarm timeout to 10
Apr 19 08:28:58 centos-8-stream-127-0-0-2-2201 systemd[1]: setroubleshootd.service: Succeeded.
```

so it smells like setroubleshoot has an idle timer which still fires when there's a pending D-Bus call? I mostly see this on RHEL 8 (setroubleshoot-plugins-3.3.14-1.el8 and setroubleshoot-server-3.3.26-5.el8). 

Same symptom in [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18661-20230418-173000-028eaf96-centos-8-stream-pybridge/TestSelinux-testTroubleshootAlerts-centos-8-stream-127.0.0.2-2801-FAIL.log.gz), there the timing information fits:

```
Apr 18 18:22:30 centos-8-stream-127-0-0-2-2201 setroubleshoot[2962]: AnalyzeThread.run(): Set alarm timeout to 10
Apr 18 18:22:39 centos-8-stream-127-0-0-2-2201 dbus-daemon[779]: [system] Activating service name='org.fedoraproject.SetroubleshootFixit' requested by ':1.37' (uid=0 pid=1454 comm="/usr/bin/python3 /usr/local/bin/cockpit-bridge --p" label="unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023") (using servicehelper)
Apr 18 18:22:40 centos-8-stream-127-0-0-2-2201 dbus-daemon[779]: [system] Successfully activated service 'org.fedoraproject.SetroubleshootFixit'
Apr 18 18:22:40 centos-8-stream-127-0-0-2-2201 org.fedoraproject.SetroubleshootFixit[3015]: ERROR:dbus.proxies:Introspect error on :1.56:/org/fedoraproject/Setroubleshootd: dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying
Apr 18 18:22:41 centos-8-stream-127-0-0-2-2201 org.fedoraproject.SetroubleshootFixit[3015]: could not attach to desktop process
Apr 18 18:22:41 centos-8-stream-127-0-0-2-2201 systemd[1]: setroubleshootd.service: Succeeded.
```

The [weather report](https://ci-weather-cockpit.apps.ocp.cloud.ci.centos.org/tests.html?days=14&repo=cockpit-project%2Fcockpit&test=test%2Fverify%2Fcheck-selinux+TestSelinux.testTroubleshootAlerts) shows that this [also](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18647-20230417-145619-d3f14f3e-fedora-38/log.html#133) [happens](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18647-20230417-081428-d3f14f3e-fedora-38/log.html#139-2) on Fedora 38.

image

**Comment by martinpitt** [comment]
When this fails locally, I now get:
```
> warning: Failed to delete alert: {"problem":null,"name":"org.freedesktop.DBus.Error.NoReply","message":"Message recipient disconnected from message bus without replying"}
```

and the error is shown in the UI:
TestSelinux-testTroubleshootAlerts-centos-8-stream-127 0 0 2-2201-FAIL

File: pkg/selinux/setroubleshoot-client.js [hunk 1]
```diff
@@ -107,69 +107,56 @@ client.init = function(capabilitiesChangedCallback) {
...
+                    doText: itm[2],
+                    analysisId: itm[3],
+                    fixable: itm[4],
+                    reportBug: itm[5],
+                }));
+                return details;
+            })
+            .catch(ex => {
+                console.warn("Unable to get alert for id", localId, ":", JSON.stringify(ex));
+                return Promise.reject(new Error(`Unable to get alert for id ${localId}: ${ex.toString()}`));
```
Comment (cockpituous) [inline pkg/selinux/setroubleshoot-client.js:135]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-100003-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/selinux/setroubleshoot-client.js [hunk 2]
```diff
@@ -107,69 +107,56 @@
...
-            client.proxyFixit.call("run_fix", [alertId, analysisId])
-                    .then(result => resolve(result[0]))
-                    .catch(ex => reject(new Error(cockpit.format(_("Unable to run fix: $0"), ex))));
-        });
-    };
+    client.runFix = (alertId, analysisId) => client.proxyFixit.call("run_fix", [alertId, analysisId])
+            .then(result => result[0])
+            .catch(ex => {
+                console.warn("Unable to run fix:", JSON.stringify(ex));
+                return Promise.reject(new Error(cockpit.format(_("Unable to run fix: $0"), ex.toString())));
```
Comment (cockpituous) [inline pkg/selinux/setroubleshoot-client.js:145]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-100003-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/selinux/setroubleshoot-client.js [hunk 3]
```diff
@@ -107,69 +107,56 @@
...
-                        console.warn("Unable to delete alert with id " + localId);
-                        console.warn(ex);
-                        reject(new Error(cockpit.format(_("Error while deleting alert: $0"), localId)));
-                    });
-        });
-    };
+    const deleteAlert = localId => client.proxy.call("delete_alert", [localId])
+            .then(success => {
+                if (!success)
+                    return Promise.reject(new Error(cockpit.format(_("Failed to delete alert: $0"), localId)));
```
Comment (cockpituous) [inline pkg/selinux/setroubleshoot-client.js:154]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-100003-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/selinux/setroubleshoot-client.js (hunk 1 above)
Comment (cockpituous) [inline pkg/selinux/setroubleshoot-client.js:135]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-114356-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/selinux/setroubleshoot-client.js (hunk 2 above)
Comment (cockpituous) [inline pkg/selinux/setroubleshoot-client.js:145]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-114356-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/selinux/setroubleshoot-client.js (hunk 3 above)
Comment (cockpituous) [inline pkg/selinux/setroubleshoot-client.js:154]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-114356-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/selinux/setroubleshoot-client.js [hunk 4]
```diff
@@ -107,69 +107,56 @@
...
+            })
+            .catch(ex => {
+                console.warn("Unable to delete alert with id", localId, ":", JSON.stringify(ex));
+                return Promise.reject(new Error(cockpit.format(_("Failed to delete alert: $0"), ex.toString())));
```
Comment (cockpituous) [inline pkg/selinux/setroubleshoot-client.js:158]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-114356-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**